modules/examples/sources/msg-generator/msg-generator-source.c

```diff
@@ -55,14 +56,23 @@ _start_timer(MsgGeneratorSource *self)
   iv_timer_register(&self->timer);
 }
 
+static void
+_start_timer_immediately(MsgGeneratorSource *self)
+{
+  iv_validate_now();
+  self->timer.expires = iv_now;
+
+  iv_timer_register(&self->timer);
+}
+
 static gboolean
 _init(LogPipe *s)
 {
   MsgGeneratorSource *self = (MsgGeneratorSource *) s;
   if (!log_source_init(s))
     return FALSE;
-
-  _start_timer(self);
+  if (self->options->max_num)
```

Comment:
Please check the same way as below (` ..max_num < 0`).

Reply:
Thanks. Fixed.